python/tvm/relay/backend/contrib/tidl.py

```diff
@@ -0,0 +1,1403 @@
+            return False
+
+        # Calibrate TIDL for the imported subgraph
+        status, out_data_q = subgraph_calibration(tidl_calib_tool, input_quant_vec, 
+                                                  input_signed, net_file, par_file)
+        if status == False:
+            return False
+        
+        # Calculate scaling factor to convert output tensor to floating point
+        # Obtain output tensor from TVM graph execution
+        output_fp = obtain_subgraph_tensor(subgraph_tensors, out_tensor_name)
+        
+        # TODO: convert following lines into a function
+        if output_fp is None:
+            return False
+        if len(output_fp) != len(out_data_q):
+            return False
+        output_signed = []
+        output_scale  = []
+        for i in range(len(output_fp)):
+            output_signed.append(int(np.amin(output_fp[i]) < 0))
+        for i in range(len(out_data_q)):
+            output_scale.append(round(out_data_q[i]/255.0,5))  # 255 is TIDL implementation specific
+        print("Output conversion: " + str(out_data_q) + ", " + str(output_scale))
+        
+        # Generate subgraph configuration file
+        subgraph_cfg_gen(artifacts_folder, subgraph_id, data_layout, 
+                         input_scale, input_signed, output_scale, output_signed)
+    return True
+
+def tensor_quant_flatten(input_tensor, data_layout):
+    r""" Convert float32 n-d array to int8 or uint8 1-d array
+    Parameters
+    ----------
+    input: float32 array 
+    data_layout: "NCHW" or "NHWC"
+    output: 1 dimension int8 or uint8 array
+    """
+
+    # only use 1 batch for calibration
+    input_tensor = input_tensor[0,:]
+    # change layout to CxHxW to use numpy.flattern to change to 1-d array
+    if data_layout == "NHWC" and len(input_tensor.shape) == 3:
+        input_tensor = input_tensor.transpose(2,0,1)
+
+    if np.amin(input_tensor) >= 0:
+        # quantize to Uint8
+        sign  = 0
+        scale = 255.0/np.amax(input_tensor)
+        quant_min = 0
+        quant_max = 255
+    else:
+        # quantize to Int8
+        sign  = 1
+        scale = 128.0/max(abs(np.amin(input_tensor)), np.amax(input_tensor))
+        quant_min = -128
+        quant_max = 127
+        
+    y = np.multiply(input_tensor, scale)
+    z = np.rint(y)
+    z = np.clip(z, quant_min, quant_max)
+    output = z.flatten()   # works only if z is in "CxHxW" format
+
+    return output, scale, sign
+
+def subgraph_calibration(calib_tool, input_quant_vec, input_signed, net_file, params_file):
+
+    # Save quantized input vector to a file for calib tool to read
+    # Saving as 'int8' or 'uint8' is the same
+    calib_raw_image = './calib_raw_data.bin'
+    if input_signed == 1:
+        input_quant_vec.astype('int8').tofile(calib_raw_image);
+    else:
+        input_quant_vec.astype('uint8').tofile(calib_raw_image);
+
+    # Prepare for calibration
+    output_tmp_file = './tempDir/precalib_net.bin'

```

Comment:
Can we use `TemporaryDirectory` for manging temp dirs?

Reply:
@trevor-m , this folder is needed when we debug TIDL import. Is it a problem to create a folder in test/python/relay folder in CI?

Reply:
We should use os.mkdir in the test

Reply:
Done